src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -439,6 +329,85 @@ private static void ValidateBuffersList(IList<ArraySegment<byte>> buffers)
             }
         }
 
+        private static void ConfigureBuffer(
+            Int32TaskSocketAsyncEventArgs saea, ArraySegment<byte> buffer, SocketFlags socketFlags, bool wrapExceptionsInIOExceptions)
+        {
+            // Configure the buffer.  We don't clear the buffers when returning the SAEA to the pool,
+            // so as to minimize overhead if the same buffer is used for subsequent operations (which is likely).
+            // But SAEA doesn't support having both a buffer and a buffer list configured, so clear out a buffer list
+            // if there is one before we set the desired buffer.
+            if (saea.BufferList != null) saea.BufferList = null;
+            saea.SetBuffer(buffer.Array, buffer.Offset, buffer.Count);
+            saea.SocketFlags = socketFlags;
+            saea._wrapExceptionsInIOExceptions = wrapExceptionsInIOExceptions;
+        }
+
+        private static void ConfigureBufferList(
+            Int32TaskSocketAsyncEventArgs saea, IList<ArraySegment<byte>> buffers, SocketFlags socketFlags)
+        {
+            // Configure the buffer list.  We don't clear the buffers when returning the SAEA to the pool,
+            // so as to minimize overhead if the same buffers are used for subsequent operations (which is likely).
+            // But SAEA doesn't support having both a buffer and a buffer list configured, so clear out a buffer
+            // if there is one before we set the desired buffer list.
+            if (saea.Buffer != null) saea.SetBuffer(null, 0, 0);
+            saea.BufferList = buffers;
+            saea.SocketFlags = socketFlags;
+        }
+
+        /// <summary>Gets a task to represent the operation.</summary>
+        /// <param name="pending">true if the operation completes asynchronously; false if it completed synchronously.</param>
+        /// <param name="saea">The event args instance used with the operation.</param>
+        /// <param name="wrapExceptionsInIOExceptions">true if SocketExceptions should be wrapped in IOExceptions; otherwise, false.</param>
+        /// <param name="isReceive">true if this is a receive; false if this is a send.</param>
+        private Task<int> GetTaskForSendReceive(bool pending, Int32TaskSocketAsyncEventArgs saea, bool wrapExceptionsInIOExceptions, bool isReceive)
+        {
+            Task<int> t;
+
+            if (pending)
+            {
+                // The operation completed asynchronously.  Get the task for the operation,
```

Comment:
Nit, but the wording here is confusing.  The operation hasn't necessarily "completed asynchronously" -- yet.  Better to say either "the operation will complete asynchronously" or just "the operation is pending".